docker/package/tezos_baking_wizard.py

```diff
@@ -0,0 +1,522 @@
+        elif self.config["key_import_mode"] == "json":
+            self.check_baker_account(
+                "sudo -u tezos tezos-client activate account baker with "
+                + self.config["json_filepath"]
+            )
+
+        else:
+            print()
+            input("Please open the Tezos Wallet app on your ledger, then hit Enter.")
+            self.check_baker_account(
+                "sudo -u tezos tezos-client import secret key baker "
+                + self.config["ledger_url"]
+            )
+
+            print()
+            input("Please open the Tezos Baking app on your ledger, then hit Enter.")
+            proc_call("sudo -u tezos tezos-client setup ledger to bake for baker")
+
+    def register_baker(self):
+        print()
+        if self.config["key_import_mode"] == "ledger":
+            input(
+                "Please open the Tezos Baking or Tezos Wallet app on your ledger, then hit Enter."
+            )
+
+        proc_call("sudo -u tezos tezos-client register key baker as delegate")
+        print(
+            "You can check a blockchain explorer (e.g. https://tzkt.io/ or https://tzstats.com/) "
+            "to see the baker status and baking rights of your account."
+        )
+
+    def start_baking(self):
+        proc_call(
+            "sudo systemctl start tezos-baking-" + self.config["network"] + ".service"
+        )
+
+
+def run_setup(cfg):
+
+    if cfg is not None:
+        setup = Setup(cfg)
+
+        print("Trying to bootstrap tezos-node")
+        setup.bootstrap_node()
+
+        if cfg["mode"] == "node":
+            return
+
+        print("Importing the baker key")
+        setup.import_baker_key()
+
+        print("Registering the baker")
+        setup.register_baker()
+
+        print("Start baking instance")
+        setup.start_baking()
+
+        print(
+            "Congratulations! All required Tezos infrastructure services should now be started."
+            " You can show logs for all the services using the 'tezos' user by running: "
+            "journalctl -f _UID=$(id tezos -u)"
+        )
+
+        if cfg["mode"] == "baker":
+            print()
+            print(
+                "To stop the baking instance, run: sudo systemctl stop tezos-baking-"
+                + cfg["network"]
+                + ".service"
+            )
+
+
+if __name__ == "__main__":
+    readline.parse_and_bind("tab: complete")
+    readline.set_completer_delims(" ")
+
+    try:
+        cfg = run_wizard()
+    except KeyboardInterrupt:
+        print()
+        print("Exiting Tezos Baking Wizard")
+        sys.exit(1)
+    except EOFError:
+        print()
+        print("Exiting Tezos Baking Wizard")
+        sys.exit(1)
+    except Exception as e:
+        print("Error in Tezos Baking Wizard, exiting.")
+        logfile = "tezos_baking_wizard.log"
+        with open(logfile, "a") as f:
+            f.write(str(e) + "\n")
+            print("The error has been logged to", logfile)
```

Comment:
Perhaps this should be out of the `with` block?

Also, I suppose it would be better if it printed the absolute path.